pkg/network/nmstate.go

```diff
@@ -72,8 +70,63 @@ func cleanUpNMState(conf *cnao.NetworkAddonsConfigSpec, ctx context.Context, cli
 
 	} else if apierrors.IsNotFound(err) {
 		// object not found, no need for action.
-		return nil
+		return []error{}
 	}
+	return []error{}
+}
+
+func removeOldStoredVersion(ctx context.Context, client k8sclient.Client) []error {
```

Comment:
I haven't went over the code, but this looks scary. We must make sure that CRs created by users are not deleted during the upgrade. Is this code safe?

Reply:
this change is only removing a part of the crd's status, that declares which stored version is used.
a similar thing is done [when droping support to deprecated versions](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning/#upgrade-existing-objects-to-a-new-stored-version)

normally there shouldn't be more than 1 storedversion AFAIK

Reply:
but maybe you're right, checking if this is necessary

Reply:
you're totally right.